src/cljs/ataru/hakija/application_hakukohde_handlers.cljs

```diff
@@ -9,19 +9,36 @@
        (filter #(= "hakukohteet" (:id %)))
        first))
 
+(defn- selected-hakukohteet [db]
+  (map :value (get-in db [:application :answers :hakukohteet :values] [])))
+
+(defn- selected-hakukohteet-and-ryhmat [db]
+  (let [selected-hakukohteet     (set (selected-hakukohteet db))
+        selected-hakukohderyhmat (->> (get-in db [:form :tarjonta :hakukohteet])
+                                      (filter #(contains? selected-hakukohteet (:oid %)))
+                                      (mapcat :hakukohderyhmat))]
+    (set (concat selected-hakukohteet selected-hakukohderyhmat))))
+
+(defn- field-intersection-with-selected-hakukohteet-and-ryhmat [db field]
+  (if-let [ids (seq (concat (get field :belongs-to-hakukohderyhma [])
```

Comment:
`(:belongs-to-hakukohderyhma field)` ja `(:belongs-to-hakukohteet field)` riittää